feat: implement local dev API endpoint management feature

limichange opened the pull request:
<img width="512" height="513" alt="image" src="https://github.com/user-attachments/assets/2ed9a995-50ec-4324-b0bb-799b74fbc24a" />


- Added ApiEndpointDialog component for managing custom API endpoints.
- Integrated API endpoint configuration into the background API service.
- Implemented methods for adding, updating, deleting, and retrieving API endpoint configurations.
- Enhanced existing request handling to support custom API endpoints.
- Updated settings configuration to include API endpoint management option.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

* **New Features**
  * Introduced API Endpoint Management in developer settings, allowing users to add, edit, enable/disable, and delete custom API endpoints via a new management dialog.
  * Added a settings menu item for API Endpoint Management under the "Dev" tab.

* **Enhancements**
  * API requests now support custom endpoint configurations when enabled in developer settings, with fallback to default endpoints.
  * Improved feedback and validation for managing custom API endpoints.

* **Bug Fixes**
  * Enhanced cache clearing for API clients to ensure updated configurations are used.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: implement API endpoint management feature

- Added ApiEndpointDialog component for managing custom API endpoints.
- Integrated API endpoint configuration into the background API service.
- Implemented methods for adding, updating, deleting, and retrieving API endpoint configurations.
- Enhanced existing request handling to support custom API endpoints.
- Updated settings configuration to include API endpoint management option.

revan-zhang commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/cab078ff-f7ab-4978-a372-7b14db9b02ab?refs=a7627de2fb543f6a412b03535b17eb17cc9cac3b&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/cab078ff-f7ab-4978-a372-7b14db9b02ab/license?refs=a7627de2fb543f6a412b03535b17eb17cc9cac3b&source=prChecksComment)

limichange committed:
feat: enhance API endpoint management with new components and functionality

- Introduced ApiEndpointForm, ApiEndpointList, and ApiEndpointListItem components for better organization and management of API endpoints.
- Added constants for service module options and labels to improve code maintainability.
- Implemented form validation and error handling for adding and editing API endpoints.
- Enhanced the API endpoint list with options to toggle, edit, and delete configurations.
- Integrated new components into the ApiEndpointDialog for a cohesive user experience.

limichange committed:
refactor: reorganize API endpoint management components

- Removed the old ApiEndpointDialog implementation and replaced it with a new structure that includes ApiEndpointForm, ApiEndpointList, and ApiEndpointListItem components for improved modularity.
- Enhanced the API endpoint management experience by integrating new components into the dialog.
- Updated the handling of API endpoint configurations to streamline the addition, editing, and deletion processes.
- Improved code organization and maintainability by separating concerns into distinct components.

limichange committed:
refactor: migrate API endpoint management to devSettings

- Removed the legacy API endpoint configuration service and integrated custom API endpoint management into the devSettings structure.
- Updated components to utilize the new devSettings for adding, editing, and deleting API endpoints.
- Enhanced the API endpoint list and form components to reflect changes in data handling and improve user experience.
- Improved code organization by consolidating related types and removing obsolete references.

limichange committed:
refactor: streamline API endpoint configuration types

- Introduced a legacy type for backward compatibility with the API endpoint configuration.
- Removed the inline legacy type definition from requestHelper.ts to improve code clarity and maintainability.
- Updated imports to utilize the new type definition from devSettings.ts, ensuring consistent type usage across the application.

limichange committed:
fix: add devSettings check for custom API endpoint configuration

- Updated the condition for returning the custom API endpoint to include a check for devSettings.enabled, ensuring that the custom configuration is only returned when both the custom config is enabled and devSettings are active.

limichange committed:
refactor: enhance endpoint retrieval logic with devSettings check

- Updated the getEndpointInfo and getEndpointDomainWhitelist functions to include a check for devSettings.enabled, ensuring that custom configurations are only used when development settings are active. This improves the flexibility and maintainability of endpoint management.

limichange committed:
refactor: remove legacy API endpoint configuration references

- Eliminated the getApiEndpointConfigPersistAtom method and its associated type to streamline the codebase, as the management of API endpoints has been migrated to the devSettings structure. This change enhances clarity and maintainability by removing obsolete code.

limichange committed:
Merge branch 'x' into api-config

limichange committed:
refactor: simplify API endpoint form validation and improve user feedback

- Removed the inline URL validation logic from the ApiEndpointForm component to streamline the code.
- Updated the ApiEndpointList component to utilize internationalization for titles and messages, enhancing user experience.
- Improved error handling and success messages for API endpoint operations by integrating localized strings.

limichange committed:
refactor: enhance ApiEndpointList component with internationalization support

- Updated the ApiEndpointList component to include the intl dependency in the useEffect and useCallback hooks, improving the handling of localized strings for better user experience.

sidmorizon committed:
Merge branch 'x' into api-config
